### PR TITLE
example(RecordLoader): Add support for a unique parameter

### PR DESCRIPTION
**Pull request by razertory**
This change will allow to load_many in the RecordLoader

### CODE REVIEW

File: examples/record_loader.rb [hunk 1]
```diff
@@ -1,18 +1,31 @@
...
   def perform(keys)
-    query(keys).each { |record| fulfill(record.public_send(@column), record) }
-    keys.each { |key| fulfill(key, nil) unless fulfilled?(key) }
+    if @unique
+      query(keys).each do |record|
+        fulfill(record.public_send(@column), record)
+      end
+      default = nil
+    else
+      query(keys).group_by(&@column).each do |key, result|
```
Comment (razertory) [inline examples/record_loader.rb]: * `group by` may not be a good choice without unique index (see: https://stackoverflow.com/questions/34115174/error-related-to-only-full-group-by-when-executing-a-query-in-mysql)

File: examples/record_loader.rb [hunk 2]
```diff
@@ -1,18 +1,31 @@
 class RecordLoader < GraphQL::Batch::Loader
-  def initialize(model, column: model.primary_key, where: nil)
+  EMPTY_ARRAY = [].freeze
+
+  def initialize(model, column: model.primary_key, where: nil, unique: true)
     @model = model
-    @column = column.to_s
-    @column_type = model.type_for_attribute(@column)
+    @column = column
+    @column_type = model.type_for_attribute(@column.to_s)
```
Comment (razertory) [inline examples/record_loader.rb]: * a `sequel model` has no `type_for_attribute` method

File: examples/record_loader.rb (hunk 1 above)
Comment (dylanahsmith) [reply examples/record_loader.rb]: This should be doing Array#group_by.  Perhaps `.to_a` should be used at the end of `query` to make this clear

File: examples/record_loader.rb (hunk 2 above)
Comment (dylanahsmith) [reply examples/record_loader.rb]: Right, that is why in https://github.com/Shopify/graphql-batch/issues/82#issuecomment-388565399 I said

> Likely the column type casting would still need to be removed to work with Sequel.

Comment (ayrton) [reply examples/record_loader.rb]: What would that look like with Sequel?

**Pull request closed without merging**